lib/core/CLoopProgress.cc

```diff
@@ -30,5 +50,40 @@ void CLoopProgress::increment(std::size_t i) {
         m_LastProgress += stride;
     }
 }
+
+void CLoopProgress::resumeRestored() {
+    this->increment(0);
```

Comment:
Does this do anything?

Reply:
It refreshes the `m_LastProgress` which *is not* persisted. This triggers us also to output all the progress we made up to the point that persist was called. I think this probably warrants some comments, also in persist.